grpclib/client.py

```diff
@@ -389,16 +395,39 @@ def __init__(self, host='127.0.0.1', port=50051, *, loop, codec=None):
                                        header_encoding='utf-8')
         self._authority = '{}:{}'.format(self._host, self._port)
 
+        if ssl and isinstance(ssl, bool):
+            ssl = self._get_default_ssl_context()
+
+        self._ssl = ssl or None
+        self._scheme = 'https' if self._ssl else 'http'
+
     def _protocol_factory(self):
         return H2Protocol(Handler(), self._config, loop=self._loop)
 
     async def __connect__(self):
         if self._protocol is None or self._protocol.handler.connection_lost:
             _, self._protocol = await self._loop.create_connection(
-                self._protocol_factory, self._host, self._port
+                self._protocol_factory, self._host, self._port,
+                ssl=self._ssl
             )
         return self._protocol
 
+    # https://python-hyper.org/projects/h2/en/stable/negotiating-http2.html
+    def _get_default_ssl_context(self):
+        if not ssl:
+            raise RuntimeError('SSL is not supported.')
+
+        ctx = ssl.create_default_context(purpose=ssl.Purpose.SERVER_AUTH)
+        ctx.options |= (ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1)
+        ctx.set_ciphers('ECDHE+AESGCM:ECDHE+CHACHA20:DHE+AESGCM:DHE+CHACHA20')
+        ctx.set_alpn_protocols(['h2', 'http/1.1'])
```

Comment:
I'm totally Ok with this, I don't worry about possible overhead, as this happens only once for a long-living connection. The only concern is that I don't think that we should advertise `http/1.1` protocol support here. Can you remove it from the list and test with your server?

Reply:
Yeah you're probably right -- let me just do a quick test with our server